<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN">
<html><head><title>R: function to test genetic association between a dichotomous trait and a batch of imputed SNPs in families using 
GEE</title>
<meta http-equiv="Content-Type" content="text/html; charset=iso-8859-1">
<link rel="stylesheet" type="text/css" href="../../R.css">
</head><body>

<table width="100%" summary="page for gee.lgst.batch.imputed {GWAF}"><tr><td>gee.lgst.batch.imputed {GWAF}</td><td align="right">R Documentation</td></tr></table>
<h2>function to test genetic association between a dichotomous trait and a batch of imputed SNPs in families using 
GEE</h2>


<h3>Description</h3>

<p>
Fit logistic regression via GEE to test association between a dichotomous phenotype 
and all imputed SNPs in a genotype file. Each family is treated as 
a cluster, with independence working correlation matrix used in the robust variance estimator.
This function applies the same trait-SNP association test to all SNPs in the imputed genotype data. 
The trait-SNP association test is carried out by <code>gee.lgst.imputed</code> function where the 
the <code>gee()</code> function from package <code>gee</code> is used.
</p>


<h3>Usage</h3>

<pre>
gee.lgst.batch.imputed(genfile, phenfile, pedfile, outfile, phen, 
covars=NULL, col.names=T, sep.ped=",", sep.phe=",", sep.gen=",")
</pre>


<h3>Arguments</h3>

<table summary="R argblock">
<tr valign="top"><td><code>genfile</code></td>
<td>
a character string naming the genotype file for reading (see format requirement in details) </td></tr>
<tr valign="top"><td><code>phenfile</code></td>
<td>
a character string naming the phenotype file for reading (see format requirement in details) </td></tr>
<tr valign="top"><td><code>pedfile</code></td>
<td>
a character string naming the pedigree file for reading (see format requirement in details) </td></tr>
<tr valign="top"><td><code>outfile</code></td>
<td>
a character string naming the result file for writing </td></tr>
<tr valign="top"><td><code>phen</code></td>
<td>
a character string for a phenotype name in phenfile </td></tr>
<tr valign="top"><td><code>covars</code></td>
<td>
a character vector for covariates in phenfile </td></tr>
<tr valign="top"><td><code>col.names</code></td>
<td>
a logical value indicating whether the output file should contain column names </td></tr>
<tr valign="top"><td><code>sep.ped</code></td>
<td>
the field separator character for pedigree file </td></tr>
<tr valign="top"><td><code>sep.phe</code></td>
<td>
the field separator character for phenotype file </td></tr>
<tr valign="top"><td><code>sep.gen</code></td>
<td>
the field separator character for genotype file </td></tr>
</table>

<h3>Details</h3>

<p>
Similar to the details for <code>gee.lgst.batch</code> but here the SNP data contains imputed genotypes (allele dosages) 
that are continuous and range from 0 to 2. In addition, the user 
specified genetic model argument is not available.
</p>


<h3>Value</h3>

<p>
No value is returned. Instead, results are written to <code>outfile</code>.
</p>
<table summary="R argblock">
<tr valign="top"><td><code>phen </code></td>
<td>
phenotype name</td></tr>
<tr valign="top"><td><code>snp </code></td>
<td>
SNP name</td></tr>
<tr valign="top"><td><code>N </code></td>
<td>
the number of individuals in analysis</td></tr>
<tr valign="top"><td><code>Nd </code></td>
<td>
the number of individuals in affected sample in analysis</td></tr>
<tr valign="top"><td><code>AF </code></td>
<td>
imputed allele frequency of coded allele</td></tr>
<tr valign="top"><td><code>AFd </code></td>
<td>
imputed allele frequency of coded allele in affected sample</td></tr>
<tr valign="top"><td><code>beta </code></td>
<td>
regression coefficient of SNP covariate</td></tr>
<tr valign="top"><td><code>se </code></td>
<td>
standard error of <code>beta</code></td></tr>
<tr valign="top"><td><code>remark </code></td>
<td>
warning or additional information for the analysis, note that the genotype counts are based
on rounded imputed genotypes; 'not converged' indicates the 
GEE analysis did not converge; 'logistic reg' indicates GEE model is replaced by logistic regression;
'exp count&lt;5' indicates any expected count is less than 5 in phenotype-genotype table; 'not converged 
and exp count&lt;5', 'logistic reg &amp; exp count&lt;5' are noted similarly; 'collinearity' indicates collinearity
exists between SNP and some covariates</td></tr>
<tr valign="top"><td><code>pval </code></td>
<td>
p-value of the chi-square statistic</td></tr>
</table>

<h3>Author(s)</h3>

<p>
Qiong Yang &lt;qyang@bu.edu&gt; and Ming-Huei Chen &lt;mhchen@bu.edu&gt;
</p>



<hr><div align="center">[Package <em>GWAF</em> version 1.2 <a href="00Index.html">Index]</a></div>

</body></html>
