<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN">
<html><head><title>R: function for making genome-wide p-value plot</title>
<meta http-equiv="Content-Type" content="text/html; charset=iso-8859-1">
<link rel="stylesheet" type="text/css" href="../../R.css">
</head><body>

<table width="100%" summary="page for GWplot {GWAF}"><tr><td>GWplot {GWAF}</td><td align="right">R Documentation</td></tr></table>
<h2>function for making genome-wide p-value plot</h2>


<h3>Description</h3>

<p>
GWplot function plots -log_{10} p-value based on SNP's chromosomal position in bitmap format.
</p>


<h3>Usage</h3>

<pre>
GWplot(data, pval, pos, chr, chr.plot = c(1:22, "X"), title.text = "", ylim = Inf, outfile, cutoff1 = 5e-08, cutoff2 = 4e-07)
</pre>


<h3>Arguments</h3>

<table summary="R argblock">
<tr valign="top"><td><code>data</code></td>
<td>
a dataframe that contains p-values, chromosome number and physical position of SNPs </td></tr>
<tr valign="top"><td><code>pval</code></td>
<td>
a character string correspong to the name of the p-value column </td></tr>
<tr valign="top"><td><code>pos</code></td>
<td>
a character string correspong to the name of column with SNP physical positions  </td></tr>
<tr valign="top"><td><code>chr</code></td>
<td>
a character string correspong to the name of column with SNP chromosome number </td></tr>
<tr valign="top"><td><code>chr.plot</code></td>
<td>
the chsomosomes of interest for GWplot; either 1:22 or c(1:22,"X"), default chr.plot=c(1:22,"X"), "X" for X chromosome </td></tr>
<tr valign="top"><td><code>title.text</code></td>
<td>
the title of the genome-wide p-value plot </td></tr>
<tr valign="top"><td><code>ylim</code></td>
<td>
the maximum of -log_{10} p-value to be plotted, useful when not want to plot extremely small p-values </td></tr>
<tr valign="top"><td><code>outfile</code></td>
<td>
the file name (xxxx.bmp) for output plot in bitmap format </td></tr>
<tr valign="top"><td><code>cutoff1</code></td>
<td>
genome-wide significance; default is 5E-8 ; p-values below this threshold will be highlighted in red</td></tr>
<tr valign="top"><td><code>cutoff2</code></td>
<td>
suggestive genome-wide significance; default is 4E-7; p-values below this threshold but above cutoff1 will be highlighted in blue </td></tr>
</table>

<h3>Details</h3>

<p>
When the dataset has 0 p-value, GWplot will generate pvalzero.csv that contain the results with 0 p-value and make the genome-wide p-value
plot by replacing 0 p-value with 5E-324. P-values that reach genome-wide significance are displayed in red color; P-values that reach 
suggestive genome-wide significance but not genome-wide significance are displayed in blue color.
</p>


<h3>Author(s)</h3>

<p>
Qiong Yang &lt;qyang@bu.edu&gt; and Ming-Huei Chen &lt;mhchen@bu.edu&gt;
</p>



<hr><div align="center">[Package <em>GWAF</em> version 1.0 <a href="00Index.html">Index]</a></div>

</body></html>
